ethcore/db/src/db.rs

```diff
@@ -93,13 +93,13 @@ pub trait Key<T> {
 /// Should be used to write value into database.
 pub trait Writable {
```

Comment:
Out of scope of this PR;

I feel like this trait should be divided into `Writeable` and `WriteableWithCache` or something.
Also, it would be nice to use `trait params` instead of repeating the trait bounds for each method in this trait for T and R. The same applies to the trait `Readable`